### DISCUSSION TITLE
email setup on unraid docker

### DISCUSSION THREAD

**Discussion by mlause** [ckulka/baikal-docker#191] [General]
Hi All,

I use an email service and was following the gmail setup guide. I'm running this in Unraid, so I added the variables in the guide as follows:

MSMTPRC_AUTH on
MSMTPRC_TLS on
MSMTPRC_TLS_TRUST_FILE /etc/ssl/certs/ca-certification.crt
etc

Here are some questions. Is this the proper way to setup the email service? How can I test it to make sure it works? The TLS trust file doesn't make sense to me. There is no location for that in the appdata folder on Unraid. Any help would be appreciated.

**Comment by ckulka** [comment]
Hi @mlause,

Those environment variables are almost correct: don't use multiple ones, there is only one environment variable `MSMTPRC` which contains the whole [msmtp configuration file content](https://marlam.de/msmtp/msmtp.html). The value of the environment variable is written to `/etc/msmtprc` when the container starts.

>How can I test it to make sure it works?

I'd say you set up a meeting to and add an invitee with an email address. That invitee should then receive an invitation by email. Alternatively (to just test msmtp and if it has been set up correctly), you can also create a new PHP file to send an email - which is what I do in the automated test to ensure msmtp works, see [cypress/fixtures/mail-test.php](https://github.com/ckulka/baikal-docker/blob/master/cypress/fixtures/mail-test.php).

>The TLS trust file doesn't make sense to me. There is no location for that in the appdata folder on Unraid.

`tls_trust_file` points to a file inside the container which comes with the container image. From what I remember, it's necessary to tell msmtp which TLS certificates to trust, since it does not use the system's default trust store. So in your case, it's to tell msmtp to trust GMail's certificate when it connects to it over TLS.

**Comment by mlause** [comment]
@ckulka

Thank you for your response. I should have said this is running as a docker under unraid. Each of those MSMTPRC variables are setup separately via a new environmental variable. So MSMTPRC_AUTH is one variable with the value set to one. Assuming that is the right way, I have it done correctly. I'd have to find out how to get the tls_trust_file from my email provider.

**Reply by ckulka** [reply]
I should have been more explicit, I’ll also update my original answer: each msmtp variable is *not* an environment variable.

The correct way is to have only one environment variable called `MSMTP`, which contains the entire content of the msmtprc file.

Upon container start, the [50-start-msmtp.sh](https://github.com/ckulka/baikal-docker/blob/cc769172bca557756f4f755e14d3293ed5b3d734/files/docker-entrypoint.d/50-start-msmtp.sh) script checks if the `MSMTP` environment variable is set. If so, then the content is written to `/etc/msmtprc`.

Regarding `tls_trusts_file`, try to use the value from the example, i.e. `tls_trust_file /etc/ssl/certs/ca-certificates.crt`. This file should exist in the container and should be enough to trust the certificate of your email provider.

**Reply by mlause** [reply]
<img width="1537" alt="image" src="https://github.com/ckulka/baikal-docker/assets/103668347/3209c57f-7a11-4f43-8f1e-6476934eac3e">

Here is my unraid config. With the MSMTP variable, nothing happens. I've trie to us on, default, and enabled for the variable. No etc/msmtprc folder is created. Should I instead create this as a container path like the rest?

**Reply by ckulka** [reply]
Ah, my mistake: the variable is called 'MSMTPRC'

**Reply by mlause** [reply]
Thanks. What should the line value be? On, true, enable, default, and leaving it blank do nothing. No etc/ folder is created.

**Comment by CamelT0E** [comment]
What are the variables exactly? 
I would like a config file or .want to use env-file.
